src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -232,6 +234,18 @@ namespace JSONHelpers
         auto result = json::from_file(save_file_path);
         if (result)
         {
+            if (!result->HasKey(L"app-zone-history"))
```

Comment:
IIRC, this will never be true, as we write empy "app-zone-history"  array if the map is empty?

Reply:
This will be true if you run the new version with code from this PR, save the settings and then restart it, because this key has been moved to another file, see `FancyZonesData::SaveFancyZonesData`